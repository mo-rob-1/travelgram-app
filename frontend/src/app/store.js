import { configureStore } from "@reduxjs/toolkit";
import authReducer from "../features/auth/authSlice";
import imageReducer from "../features/images/imageSlice";

export const store = configureStore({
  reducer: {
    auth: authReducer,
    images: imageReducer,
  },
});
