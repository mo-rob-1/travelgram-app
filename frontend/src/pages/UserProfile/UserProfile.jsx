import { useEffect, useState } from "react";
import axios from "axios";
import { useParams } from "react-router-dom";

function UserProfile() {
  const { user } = useParams();
  const [images, setImages] = useState([]);
  const [userInfo, setUserInfo] = useState([]);
  useEffect(() => {
    axios
      .get(`/api/images/${user}`)
      .then((res) => {
        console.log(res.data);
        setImages(res.data);
      })
      .catch((err) => {
        console.log(err);
      });
  }, []);

  // fetch the name of the user from the database

  useEffect(() => {
    axios
      .get(`/api/users/${user}`)
      .then((res) => {
        console.log(res.data);
        setUserInfo(res.data);
      })
      .catch((err) => {
        console.log(err);
      });
  }, []);

  return (
    <div>
      <h1>{userInfo.name} - User Profile</h1>
      {images.map((image) => (
        <div key={image._id}>
          <img src={image.image} alt={image.caption} />
          <h3>{image.caption}</h3>
        </div>
      ))}
    </div>
  );
}

export default UserProfile;
