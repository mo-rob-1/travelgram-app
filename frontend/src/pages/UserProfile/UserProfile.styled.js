import styled from "styled-components";

export const List = styled.ul`
  list-style-type: none;
  display: grid;
  grid-gap: 1rem;
  margin-top: 1.5rem;
  @media (min-width: 768px) {
    grid-template-columns: repeat(2, 1fr);
    grid-template-rows: 1fr;
    margin-top: 3rem;
  }
  @media (min-width: 1024px) {
    grid-template-columns: repeat(3, 1fr);
    gap: 1.5rem;
  }
`;

export const Figure = styled.figure`
  position: relative;
`;

export const Location = styled.figcaption`
  position: absolute;
  bottom: 0;
  padding: 1rem;
  color: #fff;
  background-color: #222222b8;
  font-size: 14px;
`;

export const IconWrapper = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
`;

export const Section = styled.section`
  margin-top: 1rem;

  @media (min-width: 1024px) {
    margin-top: 3rem;
  }
`;

export const Title = styled.h2`
  margin-top: 0;
  font-weight: 300;
`;

export const Avatar = styled.img`
  height: 100px;
  object-fit: contain;
  border-radius: 50%;
  width: 100px;
  object-fit: cover;
`;

export const UserInfoWrapper = styled.div`
  display: flex;
  justify-content: space-between;
`;

export const NumberOfImagesWrapper = styled.div`
  display: flex;
  align-items: flex-end;
`;

export const Name = styled.h4`
  display: block;
`;

export const UserInfoTextWrapper = styled.div`
  display: grid;
  gap: 1rem;
`;

export const Line = styled.div`
  height: 1px;
  background: #dfdfdf;
  margin-top: 1.5rem;
`;
