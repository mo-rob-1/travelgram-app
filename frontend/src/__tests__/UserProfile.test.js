import { render } from "@testing-library/react";
import { createMemoryHistory } from "history";
import { Provider } from "react-redux";
import { store } from "../app/store";
import { BrowserRouter as Router } from "react-router-dom";
import UserProfile from "../pages/UserProfile/UserProfile";

describe("UserProfile", () => {
  it("renders the user profile page", () => {
    const history = createMemoryHistory();
    const { asFragment } = render(
      <Provider store={store} history={history}>
        <Router>
          <UserProfile />
        </Router>
      </Provider>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  // it renders the no images text if there are no images
  it("renders the no images text if there are no images", () => {
    const history = createMemoryHistory();
    const { getByTestId } = render(
      <Provider store={store} history={history}>
        <Router>
          <UserProfile />
        </Router>
      </Provider>
    );
    const text = getByTestId("no-images");
    expect(text).toBeTruthy();
  });
});
