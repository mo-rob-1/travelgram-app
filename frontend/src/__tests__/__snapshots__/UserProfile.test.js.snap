// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserProfile renders the user profile page 1`] = `
<DocumentFragment>
  <div>
    <div>
      <h1>
         - My Images
      </h1>
      <h3
        data-testid="no-images"
      >
        This user has not uploaded any images yet.
      </h3>
    </div>
  </div>
</DocumentFragment>
`;
